viewing_party/main.py

```diff
@@ -0,0 +1,142 @@
+    for entry in movies_user_watched:
+        if entry not in movies_friends_watched:
+            unique_titles.append(entry)
+    return unique_titles
+
+def get_friends_unique_watched(user_data):
+    """Determine and return a list of movies that the user's friends
+        have watched but that the user has not watched.
+        """
+    user_watched = []
+    friends_watched = []
+    return_watched = []
+    for film in user_data["watched"]:
+        user_watched.append(film)
+    for friend in user_data["friends"]:
+        for friend_data in friend["watched"]:
+            friends_watched.append(friend_data)
+    # list comprehension to address duplicate dicts       
+    dry_friends_watched = [i for n, i in enumerate(friends_watched) if i not in friends_watched[n + 1:]]
+    for relevant_movie in dry_friends_watched:
+        if relevant_movie not in user_watched:
+            return_watched.append(relevant_movie)
+    return return_watched
+
+def get_available_recs(user_data):
+    """Determine and return a list of movies that the user has not watched,
+        that at least one of the user's friends has watched and that is
+        available to the user given their subscriptions.
+        """
+    recommended_movies = []
+    for friend_watch_list in user_data["friends"]: 
+        for movie_host_detail in friend_watch_list["watched"]:
+            if (movie_host_detail not in user_data["watched"]) and (movie_host_detail["host"] in user_data["subscriptions"]):
+                recommended_movies.append(movie_host_detail)
+    # address duplicates with list comprehension
+    dry_recommended = [i for n, i in enumerate(recommended_movies) if i not in recommended_movies[n + 1:]]
+    return dry_recommended
+
+def get_new_rec_by_genre(user_data):
+    """Determine and return a list of movie recommendations that the user
+        has not watched, that at least one of the user's friends has watched,
+        and whose genre matches the user's favorite genre.
+        """
+    genre_for_compare = get_most_watched_genre(user_data)
+    recs_by_genre = []
+    for friend_watched_list in user_data["friends"]:
+        for friend_movie in friend_watched_list["watched"]: 
+            if (friend_movie not in user_data["watched"]) and (friend_movie["genre"] == genre_for_compare):
+                recs_by_genre.append(friend_movie)
+    return recs_by_genre
+
+def get_rec_from_favorites(user_data):
+    """Determine and return a list of movie recommendations that none of the
+        user's friends have seen and that is in the user's 'FAVORITES' list.
+        """
+    hold_friend_watched = []
+    hold_my_watched = []
+    for friend_watched_list in user_data["friends"]:
+        for friend_movie_watched in friend_watched_list["watched"]: 
+            hold_friend_watched.append(friend_movie_watched)
+    for my_movie in user_data["watched"]:
+        if my_movie in user_data["favorites"]:
+            hold_my_watched.append(my_movie) 
```

Comment:
Why not only add the movie to `hold_my_watched` if the movie isn't in `hold_friend_watched`?